src/table/use-sticky-columns.tsx

```diff
@@ -0,0 +1,230 @@
+const MINIMUM_SCROLLABLE_SPACE = 148;
+
+const useCellOffsets = (tableCellRefs: Array<React.RefObject<HTMLTableCellElement>>) => {
+  // This hook calculates the cumulative offsets of the cells in each column of the table, based on the widths of their siblings
+
+  const [cellOffsets, setCellOffsets] = useState<CellOffsets>({ first: [], last: [] });
+
+  const calculateOffsetWidths = (widths: number[]) => {
+    return widths.map((elem, index) => widths.slice(0, index + 1).reduce((a, b) => a + b));
+  };
+
+  const updateCellOffsets = useCallback(() => {
+    // Calculate widths of all previous siblings of each table cell in the `tableCellRefs` array
+    const firstColumnsOffsets = [
+      0,
+      ...tableCellRefs.map(ref => (ref?.current?.previousSibling as HTMLTableCellElement)?.offsetWidth).filter(Boolean),
+    ];
+    const cumulativeFirstColumnsWidths = calculateOffsetWidths(firstColumnsOffsets);
+
+    // Calculate widths of all next siblings of each table cell in the `tableCellRefs` array
+    const lastColumnsOffsets = [
+      ...tableCellRefs.map(ref => (ref?.current?.nextSibling as HTMLTableCellElement)?.offsetWidth).filter(Boolean),
+      0,
+    ];
+    const cumulativeLastColumnsWidths = calculateOffsetWidths(lastColumnsOffsets).reverse();
+
+    setCellOffsets({ first: cumulativeFirstColumnsWidths, last: cumulativeLastColumnsWidths });
+  }, [tableCellRefs]);
+
+  useLayoutEffect(() => {
+    updateCellOffsets();
+  }, [updateCellOffsets]);
+
+  return { cellOffsets, updateCellOffsets };
+};
+
+interface GetStickyClassNamesProps {
+  styles: { [key: string]: string };
+  isSticky: boolean;
+  isLastStickyLeft: boolean;
+  isLastStickyRight: boolean;
+  isStuckToTheLeft: boolean;
+  isStuckToTheRight: boolean;
+}
+
+export function getStickyClassNames({
+  styles,
+  isSticky,
+  isLastStickyLeft,
+  isLastStickyRight,
+  isStuckToTheLeft,
+  isStuckToTheRight,
+}: GetStickyClassNamesProps) {
+  return clsx({
+    [styles['sticky-cell']]: isSticky,
+    [styles['sticky-cell-last-left']]: isStuckToTheLeft && isLastStickyLeft,
+    [styles['sticky-cell-last-right']]: isStuckToTheRight && isLastStickyRight,
+  });
+}
+
+export const useStickyColumns = ({
+  containerWidth,
+  hasSelection,
+  isWrapperScrollable,
+  stickyColumns,
+  tableRefObject,
+  visibleColumnsLength,
+}: StickyColumnParams) => {
+  // Check if there are any sticky columns
+  const noStickyColumns = !stickyColumns || (stickyColumns.first === 0 && stickyColumns.last === 0);
+
+  const [shouldDisable, setShouldDisable] = useState<boolean>(noStickyColumns);
+  const [tableCellRefs, setTableCellRefs] = useState<Array<React.RefObject<HTMLTableCellElement>>>([]);
+  const { cellOffsets, updateCellOffsets } = useCellOffsets(tableCellRefs);
+
+  // Compute left table padding
+  const table = tableRefObject.current;
+  const tableLeftPadding = table ? Number(getComputedStyle(table).paddingLeft.slice(0, -2)) : 0;
```

Comment:
I think this can be simplified with `parseInt(getComputedStyle(table).paddingLeft)`